File: refactored/assertion-roulette/CSharpXUnit/CSharpXUnit.Tests/Tests/Should_transform_underscores_to_spaces_when_required_headers_have_spaces.cs
using Xunit;

public class ExternalDataSpecificationTests
{
    [Fact]
    public void ShouldTransformUnderscoresToSpacesWhenRequiredHeadersHaveSpaces()
    {
        {
            var sut = CreateSut(new Dictionary<string, string> { { "product_name", "product" } });
            var result = sut.GetExampleRecords(new string[] { "product name" });
            Assert.NotNull(result, "Explanation message");
            Assert.Equal(3, result.GetItems().Count, "Explanation message");
            Assert.True(result.GetHeader().Contains("product name"), "Explanation message");
            Assert.Equal("Chocolate", result.GetItems()[0].GetFields()["product name"].GetValue(), "Explanation message");
        }
    }
}

File: refactored/assertion-roulette/python-pytest/Should_transform_underscores_to_spaces_when_required_headers_have_spaces.py
import pytest

class TestExternalDataSpecification:

    def test_should_transform_underscores_to_spaces_when_required_headers_have_spaces(self):
        sut = create_sut({"product_name": "product"})
        result = sut.get_example_records(["product name"])
        assert result is not None, "Explanation message"
        assert len(result.get_items()) == 3, "Explanation message"
        assert "product name" in result.get_header(), "Explanation message"
        assert result.get_items()[0].get_fields()["product name"].get_value() == "Chocolate", "Explanation message"
